tests/components/homekit_controller/test_init.py

```diff
@@ -27,3 +30,24 @@ async def test_unload_on_stop(hass, utcnow):
         await hass.async_block_till_done()
 
     assert async_unlock_mock.called
+
+
+async def test_async_remove_entry(hass: HomeAssistant):
+    """Test unpairing a component."""
+    helper = await setup_test_component(hass, create_motion_sensor_service)
+
+    hkid = "00:00:00:00:00:00"
+
+    with patch("aiohomekit.Controller") as controller_cls:
+        # Setup a fake controller with 1 pairing
+        controller = controller_cls.return_value = FakeController()
+        await controller.add_paired_device([helper.accessory], hkid)
+        assert len(controller.pairings) == 1
+
+        assert hkid in hass.data[ENTITY_MAP].storage_data
```

Comment:
Mock storage instead and assert the contents of that. `hass.data` is a detail but storage is persistent state that we can assert.

https://github.com/home-assistant/core/blob/63f8e437ed0bf79d72286853b7c2e7c01abef91f/tests/conftest.py#L231